src/main/kotlin/org/partiql/scribe/sql/RexConverter.kt

```diff
@@ -1,448 +1,595 @@
-     *
-     * NOTE: [Rex.Op.TupleUnion]'s get constant folded in the [org.partiql.planner.typer.PlanTyper].
-     *
-     * Example:
-     * ```
-     * SELECT t.* FROM t
-     * -- Gets converted into
-     * SELECT VALUE TUPLEUNION({ 'a': <INT>, 'b': <DECIMAL> }) FROM t
-     * -- Gets constant folded (in the PlanTyper) into:
-     * SELECT VALUE { 'a': <INT>, 'b': <DECIMAL> } FROM t
-     * -- Gets converted into:
-     * SELECT VALUE $__x
-     * -> PROJECT < $__x: TUPLEUNION(...) >
-     * -> SCAN t
-     * -- Gets converted into:
-     * SELECT a, b, c FROM t
-     * -- Equivalent
-     * SELECT "t".a, "t.b, "t".c FROM t AS "t"
-     * ```
-     *
-     * If unable to convert into SQL-style projections (due to open content structs, non-struct arguments, etc), we
-     * return null.
-     */
-    @OptIn(PartiQLValueExperimental::class)
-    private fun convertSelectValueToSqlSelect(
-        curr: Select?,
-        constructor: Rex,
-        input: Rel,
-        setq: SetQuantifier?,
-    ): Select? {
-        val relProject = input.op as? Rel.Op.Project ?: return null
-        val structOp = getConstructorFromProjection(constructor, relProject)?.op as? Rex.Op.Struct ?: return null
-        val newRexConverter = RexConverter(transform, Locals(relProject.input.type.schema))
-        val type = constructor.type as? StructType ?: return null
-        if (type.constraints.contains(TupleConstraint.Open(false))
-                .not() || type.constraints.contains(TupleConstraint.Ordered).not()
-        ) {
-            return null
-        }
+    override fun visitSubqueryComp(
+        rex: RexSubqueryComp,
+        ctx: Unit,
+    ): Expr {
+        listener.reportAndThrow(
+            ScribeProblem.simpleError(
+                code = ScribeProblem.UNSUPPORTED_PLAN_TO_AST_CONVERSION,
+                message = "SUBQUERYCOMP is not yet supported",
+            ),
+        )
+    }
 
-        // AGG HACK; this is terrible!
-        val projections = if (curr != null && curr is Select.Project) {
-            val first = curr.items.first() as Select.Project.Item.Expression
-            val struct = first.expr as Expr.Struct
-            struct.fields.map {
-                val expr = it.value
-                val asAlias = binder(((it.name as Expr.Lit).value as StringValue).value ?: "")
-                selectProjectItemExpression(expr, asAlias)
-            }
-        } else {
-            structOp.fields.map { field ->
-                val key = field.k.op
-                if (key !is Rex.Op.Lit || key.value !is StringValue) {
-                    return null
-                }
-                val fieldName = (key.value as StringValue).value ?: return null
-                //
-                val expr = newRexConverter.apply(field.v)
-                val asAlias = binder(fieldName)
-                selectProjectItemExpression(expr, asAlias)
-            }
-        }
+    override fun visitSubqueryIn(
```

Comment:
`RexSubqueryIn`, `RexSubqueryComp`, & `RexSubqueryTest` are abstract base classes. Should they be here?

Reply:
All of the `Rex`s and `Rel`s in the plan are abstract classes, so they should still be here